client/client.go

```diff
@@ -169,28 +240,19 @@ func newClient(creds *identity.Credentials, auth_method identity.AuthMode, httpC
 		client_creds.URL = client_creds.URL + apiTokens
 	}
 	client := authenticatingClient{
-		creds:                       &client_creds,
-		requiredServiceTypes:        defaultRequiredServiceTypes,
-		client:                      client{logger: logger, httpClient: httpClient},
+		creds:                &client_creds,
+		requiredServiceTypes: defaultRequiredServiceTypes,
+		client: client{
+			logger:     logger,
+			httpClient: httpClient,
+		},
 		apiVersionDiscoveryDisabled: set.NewStrings(),
 	}
 	client.auth = &client
 	client.authMode = identity.NewAuthenticator(auth_method, httpClient)
 	return &client
 }
 
-func NewClient(creds *identity.Credentials, authMethod identity.AuthMode, logger logging.CompatLogger) AuthenticatingClient {
-	return newClient(creds, authMethod, sharedHttpClient, logger)
-}
-
-func NewNonValidatingClient(creds *identity.Credentials, authMethod identity.AuthMode, logger logging.CompatLogger) AuthenticatingClient {
-	return newClient(creds, authMethod, goosehttp.NewNonSSLValidating(), logger)
-}
-
-func NewClientTLSConfig(creds *identity.Credentials, authMethod identity.AuthMode, logger logging.CompatLogger, config *tls.Config) AuthenticatingClient {
-	return newClient(creds, authMethod, goosehttp.NewWithTLSConfig(config), logger)
-}
-
```

Comment:
I've just congregated them together with the other constructors to highlight the problem at hand.